### PR TITLE
feat: avm/res/app/managed-environment

### PR DESCRIPTION
**Pull request by arnoldna**
Migrated managed-environment to AVM

[avm.res.app.managed-environment](https://github.com/arnoldna/bicep-registry-modules/actions/workflows/avm.res.app.managed-environment.yml)

### CODE REVIEW

File: avm/utilities/pipelines/staticValidation/psrule/.ps-rule/min-suppress.Rule.yaml [hunk 1]
```diff
@@ -23,6 +23,8 @@ spec:
     - Azure.TrafficManager.Protocol
       # Azure Load Balancer specific
     - Azure.LB.Probe
+      # Azure Container App specific
```
Comment (AlexanderSehr) [inline avm/utilities/pipelines/staticValidation/psrule/.ps-rule/min-suppress.Rule.yaml:26]:
```suggestion
      # Azure Managed-environment specific
```
While the test name doens't give it away, I think it makes sense to connect this to the module in which it is relevant, that is, managed-environment.

File: avm/res/app/managed-environment/main.bicep [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+output resourceId string = managedEnvironment.id
+
+@description('The Default domain of the Managed Environment.')
+output defaultDomain string = managedEnvironment.properties.defaultDomain
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+type diagnosticSettingType = {
```
Comment (AlexanderSehr) [inline avm/res/app/managed-environment/main.bicep:177]: This does not seem to be used

Comment (ChrisSidebotham) [reply avm/res/app/managed-environment/main.bicep:177]: This should be removed as `managedEnvironment` has its own properties to support console & system logs

File: avm/res/app/managed-environment/tests/e2e/max/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+param serviceShort string = 'amemax'
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableTelemetry bool = true
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+@description('Optional. WorkloadProfile')
+param workloadProfiles array = [
```
Comment (AlexanderSehr) [inline avm/res/app/managed-environment/tests/e2e/max/main.test.bicep:29]: Why is this a parameter?

File: avm/res/app/managed-environment/tests/e2e/waf-aligned/main.test.bicep [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+param serviceShort string = 'amewaf'
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableTelemetry bool = true
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+@description('Optional. WorkloadProfile')
+param workloadProfiles array = [
```
Comment (AlexanderSehr) [inline avm/res/app/managed-environment/tests/e2e/waf-aligned/main.test.bicep:29]: Also here - this should probably not be a parameter

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Is missing the update to the codeowners file (and group creation) if I'm not mistaken

File: avm/res/app/managed-environment/tests/e2e/max/main.test.bicep (hunk 3 above)
Comment (arnoldna) [reply avm/res/app/managed-environment/tests/e2e/max/main.test.bicep:29]: A workload profile determines the amount of compute and memory resources available to the container apps deployed in an environment. Profiles are configured to fit the different needs of your applications.When you created the managed environment you can set profiles that can be used by your container apps.

https://learn.microsoft.com/en-us/azure/templates/microsoft.app/2023-05-01/managedenvironments?pivots=deployment-language-bicep#workloadprofile
https://learn.microsoft.com/en-us/azure/container-apps/workload-profiles-overview

Comment (AlexanderSehr) [reply avm/res/app/managed-environment/tests/e2e/max/main.test.bicep:29]: Right right @arnoldna  - but why is this a parameter of the test file?

Comment (arnoldna) [reply avm/res/app/managed-environment/tests/e2e/max/main.test.bicep:29]: My understanding was that the max tests were to deploy all of the potential configuration options. If that is the case then these parameters are there to ensure that someone could deploy this if necessary. I could see removing it from WAF if this parameter isn't part of the framework for managed environments.

File: avm/res/app/managed-environment/tests/e2e/max/main.test.bicep [hunk 5]
```diff
@@ -69,13 +56,20 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
...
+        maximumCount: 3
+      }
+    ]
     internal: true
     dockerBridgeCidr: '172.16.0.1/28'
     platformReservedCidr: '172.17.17.0/24'
     platformReservedDnsIP: '172.17.17.17'
     infrastructureSubnetId: nestedDependencies.outputs.subnetResourceId
-    infrastructureResourceGroupName: infrastructureResourceGroupName
+    infrastructureResourceGroupName: 'me-dep-${namePrefix}-app.managedenvironments-${serviceShort}-rg'
```
Comment (AlexanderSehr) [inline avm/res/app/managed-environment/tests/e2e/max/main.test.bicep:72]:
```suggestion
    infrastructureResourceGroupName: 'me-${resourceGroupName}'
```

File: avm/res/app/managed-environment/tests/e2e/waf-aligned/main.test.bicep [hunk 6]
```diff
@@ -68,13 +56,20 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
...
+        maximumCount: 3
+      }
+    ]
     internal: true
     dockerBridgeCidr: '172.16.0.1/28'
     platformReservedCidr: '172.17.17.0/24'
     platformReservedDnsIP: '172.17.17.17'
     infrastructureSubnetId: nestedDependencies.outputs.subnetResourceId
-    infrastructureResourceGroupName: infrastructureResourceGroupName
+    infrastructureResourceGroupName: 'me-dep-${namePrefix}-app.managedenvironments-${serviceShort}-rg'
```
Comment (AlexanderSehr) [inline avm/res/app/managed-environment/tests/e2e/waf-aligned/main.test.bicep:72]:
```suggestion
    infrastructureResourceGroupName: 'me-${resourceGroupName}'
```

File: avm/res/app/managed-environment/tests/e2e/max/main.test.bicep [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    location: location
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
```
Comment (AlexanderSehr) [inline avm/res/app/managed-environment/tests/e2e/max/main.test.bicep:51]: Missing role assignment test

File: avm/res/app/managed-environment/tests/e2e/max/main.test.bicep [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+param resourceGroupName string = 'dep-${namePrefix}-app.managedenvironments-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'amemax'
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableTelemetry bool = true
```
Comment (AlexanderSehr) [inline avm/res/app/managed-environment/tests/e2e/max/main.test.bicep:20]: One last one - please remove this parameter from all test files

Comment (AlexanderSehr) [reply avm/res/app/managed-environment/tests/e2e/max/main.test.bicep:20]: Requires a regeneration of the readme

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**